src/app/pages/Home/Home.tsx

```diff
@@ -36,24 +32,26 @@ export default function Home(): JSX.Element {
         onSubmit={searchForDoctors}
       />
       <section className={styles.page__section}>
-        {errorMessage ? <h3>{errorMessage}</h3> : ''}
-        {doctors.map((doctor) => {
-          return (
-            <DoctorCard
-              className={styles.card}
-              id={doctor.id}
-              key={doctor.id}
-              imgPath={doctor.image}
-              title={doctor.title}
-              name={doctor.name}
-              surename={doctor.surename}
-              adress={doctor.adress}
-              plz={doctor.plz}
-              city={doctor.city}
-              isShowButton={true}
-            />
-          );
-        })}
+        {searchValue &&
+          doctors
+            .filter((doctor) => matchSearch(doctor, searchValue))
+            .map((doctor) => {
+              return (
+                <DoctorCard
+                  className={styles.card}
+                  id={doctor.id}
+                  key={doctor.id}
+                  imgPath={doctor.image}
+                  title={doctor.title}
+                  name={doctor.name}
+                  surename={doctor.surename}
+                  adress={doctor.adress}
+                  plz={doctor.plz}
+                  city={doctor.city}
+                  isShowButton={true}
+                />
+              );
+            })}
```

Comment:
The problem is that you immediately render your search results on input, but the actual query is only set when the form is submitted. That way I could search, find and see details for doctors without setting a query, which might result in unexpected search results once I go back